torch/csrc/distributed/c10d/Backend.hpp

```diff
@@ -418,6 +436,13 @@ class TORCH_API Backend : public torch::CustomClassHolder {
         c10::str("Backend ", getBackendName(), " does not support getError"));
   }
 
+  virtual uint64_t getCommSplitCounter() const {
```

Comment:
Does this need to differ on a backend basis or can we just have a simple impl in ProcessGroup. We could rename this just `incrementSplitCount()` and have it increment and return the value.